pages/about/news/[slug].vue

```diff
@@ -0,0 +1,183 @@
+<!-- eslint-disable no-console -->
+
+<script setup>
+// HELPERS
+import _get from 'lodash/get'
+import format from 'date-fns/format'
+import removeTags from '../utils/removeTags'
+
+// GQL
+import ARTICLE_DETAIL from '../gql/queries/ArticleDetail.gql'
+
+const { $graphql } = useNuxtApp()
+
+const { data } = await useAsyncData('news-story', async () => {
+  const data = await $graphql.default.request(ARTICLE_DETAIL, { slug: params.slug })
+
+  if (!data.entry) {
+    error({ statusCode: 404, message: 'Page not found' })
+  }
+
+  // Elastic search?
+  // if (data) {
+  //   if (data.entry) {
+  //     data.entry.articleCategory = data.entry.category
+  //     delete data.entry.category
+
+  //     await $elasticsearchplugin.index(data.entry, params.slug)
+  //     data.entry.category = data.entry.articleCategory
+  //   }
+  //   console.log(
+  //     'News Data fetched: ' + JSON.stringify(data.entry.category)
+  //   )
+  // }
+
+  return data
+})
+console.log('hello')
+console.log(data.value)
+
+const page = ref(_get(data.value, 'entry', {}))
+
+console.log('goodbye')
+console.log(page.value)
+
+useHead({
+  title: page.value ? page.value.title : '... loading',
+  meta: [
+    {
+      hid: 'description',
+      name: 'description',
+      content: removeTags(page.value.text)
+    }
+  ]
+})
+
+// const parsedByline = computed(() => {
+//   const byline = (page.value.contributors || []).map((contributor) => {
```

Comment:
In the original code, this byline map expression was missing a return value if the condition was not met, and the linter gave a warning about it. I added 'contributor' as the return value. Can you confirm if that's accurate?

Reply:
We cannot use contributor, as it is an object  since parsebyline should return a array of string. I think we should use reduce instead of map in this situation , By using reduce there will no undefined or empty objects in the array
parsedByline() {
  const contributors = this.page.contributors || [];
  const byline = contributors.reduce((acc, contributor) => {
    const hasStaffMember = contributor.staffMember && contributor.staffMember.length > 0;
    const hasTitle = !!contributor.title;
    if (hasStaffMember || hasTitle) {
      const title = contributor.title || contributor.staffMember[0].title;
      const bylineText = `${contributor.byline || ''} ${title}`;
      acc.push(bylineText);
    }
    return acc;
  }, []);

  return byline;
}